### PR TITLE
Add test case for dynamic message routing

### PR DESCRIPTION
**Pull request by ThishaniLucas**

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.wso2.carbon.esb.message.store.test;
```
Comment (manoramahp) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: Add license header

Comment (ThishaniLucas) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: done

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        super.init();
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Derive the store name from the request and route it dynamically")
+    public void testDynamicRouting() throws Exception {
+        OMElement response = axis2Client
+                .sendSimpleStockQuoteRequest(getProxyServiceURLHttp("storeDynamicRoutingTestProxy"), null,
+                        "dynamicRoutingMessageStore");
+        assertTrue(response.toString().contains("dynamicRoutingMessageStore"), "Message not stored!");
```
Comment (himasha) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: Can we add a more descriptive assert message?

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Derive the store name from the request and route it dynamically")
+    public void testDynamicRouting() throws Exception {
+        OMElement response = axis2Client
+                .sendSimpleStockQuoteRequest(getProxyServiceURLHttp("storeDynamicRoutingTestProxy"), null,
```
Comment (himasha) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: Where/when are we loading 'storeDynamicRoutingTestProxy' ?

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+/**
+ * This test case tests whether a message can be routed dynamically by
+ * deriving the store name from the request.
+ */
+public class DynamicMessageRoutingTestCase extends ESBIntegrationTest {
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+    }
```
Comment (manoramahp) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: It's better to verify the proxy service existence here by invoking,
verifyProxyServiceExistence("storeDynamicRoutingTestProxy");

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * This test case tests whether a message can be routed dynamically by
+ * deriving the store name from the request.
+ */
+public class DynamicMessageRoutingTestCase extends ESBIntegrationTest {
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
```
Comment (manoramahp) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: It's better to verify the proxy service existence here by invoking,
verifyProxyServiceExistence("storeDynamicRoutingTestProxy");

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java (hunk 3 above)
Comment (ThishaniLucas) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: Proxy is loaded at the time of server start. This is according to the new integration tests structure.

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java (hunk 2 above)
Comment (ThishaniLucas) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: done

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java (hunk 5 above)
Comment (ThishaniLucas) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/store/test/DynamicMessageRoutingTestCase.java]: done

**Pull request merged**